share/hooklib/juju.py

```diff
@@ -0,0 +1,24 @@
+from subprocess import run, PIPE
+import yaml
+
+
+def status():
+    """ Get juju status
+    """
+    sh = run('juju status --format yaml', shell=True, stdout=PIPE)
+    return yaml.load(sh.stdout.decode())
+
+
+def leader(application):
+    """ Grabs the leader of a set of application units
+
+    Arguments:
+    application: name of application to query.
+    """
+    sh = run('juju run --application $1 is-leader --format yaml',
+             shell=True, stdout=PIPE)
+    leader_yaml = yaml.load(sh.stdout.decode())
```

Comment:
can you add a check for success and log an error message if the run() fails? as-is this will send a crappy yaml traceback up the chain.

up to you if you want to then raise a new exception subclass like 'JujuException' or just return None or something